### ISSUE TITLE
Some help with assigning N-N relationship

### ISSUE THREAD

**Issue by Slowylowy** [demianrasko/Dynamics-365-Workflow-Tools#99]
I have some issues using the Associate Entity method. 

I want to associate a specific Portal Web Role to a Contact by using a workflow (so I can do the assignment in bulk). For this I create an on demand workflow within the context of Contact. I add a step with the Associate Entity custom activity and use these input parameters:

- Relationship name: adx_webrole_contact
- Relationship entity name: adx_webrole_contact (both Name and Entity Name are the same)
- Record URL: https://<org>.crm4.dynamics.com/main.aspx?etn=adx_webrole&id={32C6733F-6241-E811-A82F-000D3AB4B197}&pagetype=entityrecord (direct URL to the specific web role I want to associate)

I receive the error "Unexpected exception: Input string was not in a correct format.". So I assume that the input parameters are not correct. 

Would be great if you could help me out!

Regards and thanks for the great tools!
Elowy.

**Comment by ovinophile** [comment]
The Record URL mus have parameters for "etc=" and "id=" (etc is entity type code). The format should be similar to what you get when using the "Record URL(Dynamic)" attribute for an entity in a workflow.

"Record URL(Dynamic)" also doesn't include braces around the id - if you're still getting an error, try it without them.

**Comment by Slowylowy** [comment]
Thanks! This record URL worked well: https://<site>.crm4.dynamics.com/main.aspx?etc=10016&extraqs=formid=32d92d2c-67d6-4c8c-b406-6ac64391f89a&id=32C6733F-6241-E811-A82F-000D3AB4B197&pagetype=entityrecord

**Comment by marealtors** [comment]
I'm not getting an error, I have the url encoded properly (I think, see below) and it saves the workflow and activates it properly, but when I run it on a given contact, it doesn't associate the two records. The workflow completes with a status of "success." I've tried it with a couple different entities as the "destination" entity to associate the entity specified below with, always same result - nothing happens, no failures,, but the association is not created. Specifically I've tried it with Contacts and Meetings, and the entity below is Tags. I believe Meetings and Tags are custom entities in our system, but they have existing N:N relationships and can be associated manually as usual. Is this limited only to built-in CRM entities, maybe?

https://my.crmurl.org/main.aspx?etc=10095&extraqs=formid=e68ff677-accd-4ea5-8ca0-c1477981192b&id=86adb0f7-9a1b-e711-85ac-00155d24051f&pagetype=entityrecord
